library-best-practices/JLBP-16.md

```diff
@@ -0,0 +1,60 @@
+[JLBP-16] Ensure version alignment of dependencies for consumers
+----------------------------------------------------------------
+
+For multi-module projects, this best practice assumes you have already applied
+[JLBP-15](JLBP-15.md), so that your project has a BOM.
+
+### Achieving version alignment
+
+- Version alignment is defined in the [glossary](glossary.md).
+- Version alignment increases the likelihood that consumers' build systems will
+  pick the right versions of direct and transitive dependencies, reducing the
+  number of conflicts.
+- As noted in the definition, to fix a version misalignment caused by a shorter
+  path to a version of a dependency that is not the upper bound in the
+  dependency tree, a direct dependency needs to be added so that Maven consumers
+  will pick the correct version.
+- See the details specific to each build system in the sections below.
+
+### Maven
+
+- Use `requireUpperBoundDeps` enforcement to ensure that you are using the
+  version of each dependency that is the highest in your dependency tree.
+- To ensure that dependencies between modules in the project are consistent,
+  have the parent POM import the library's own BOM into its
+  `<dependencyManagement>` section.
+  - [Example import in google-cloud-java](https://github.com/GoogleCloudPlatform/google-cloud-java/blob/36409f5b1df89609eaef92d09cebea97931339bd/google-cloud-clients/pom.xml#L174).
+- To ensure that usage of dependencies is consistent, for any direct
+  dependencies that have a BOM of their own, import those BOMs in a
+  `<dependencyManagement>` section.
+  - If multiple imported BOMs manage the same dependency, use the
+    `<dependencyManagement>` of the parent POM to select a compatible version,
+    typically the highest.
+    - Order doesn't matter for the override, because all explicit version
+      declarations in dependencyManagement take precedence over all BOM
+      imports. (Order between BOMs matters - the earliest import of a dependency
+      takes precedence over later imports of the same dependency.)
+- For direct dependencies that don't have a BOM (for example, Joda-Time), make
+  sure there is only one location to define the version, so that the library
```

Comment:
make sure only one pom.xml defines the version

Reply:
Will fix.